src/pages/events.tsx

```diff
@@ -0,0 +1,176 @@
+              placeholder="Search for events by name"
+              value={searchField}
+              onChange={handleChange}
+            />
+          </div>
+          <div
+            className={`bg-events-card-bg p-2 h-[46px] rounded-lg flex-row justify-center items-center space-x-1 px-20 shrink hidden lg:flex cursor-pointer ${
+              filterState === filterStates.all ? "!bg-events-baby-blue" : ""
+            }`}
+            onClick={() => handleUiClick(filterStates.all)}
+          >
+            <ListIcon height={20} width={20} className={`${filterState === filterStates.all ? "text-events-user-card-bg fill-current" : ""}`} />{" "}
+            <p className={`${filterState === filterStates.all ? "text-events-user-card-bg" : ""} text-nowrap`}>All Events</p>
+          </div>
+          <div
+            className={`bg-events-card-bg p-2 h-[46px] rounded-lg flex-row justify-center items-center space-x-1 px-20 shrink hidden lg:flex cursor-pointer ${
+              filterState === filterStates.saved ? "!bg-events-baby-blue" : ""
+            }`}
+            onClick={() => handleUiClick(filterStates.saved)}
+          >
+            <Bookmark height={20} width={20} className={`${filterState === filterStates.saved ? "text-events-user-card-bg fill-current" : ""}`} />
+            <p className={`${filterState === filterStates.saved ? "text-events-user-card-bg" : ""}`}>Saved</p>
+          </div>
+        </div>
+        <div className="flex flex-row space-x-3 mb-6 lg:hidden">
+          <div
+            className={`bg-events-card-bg p-2 h-[46px] rounded-lg flex-row justify-center items-center space-x-1 shrink flex grow lg:hidden cursor-pointer ${
+              filterState === filterStates.all ? "!bg-events-baby-blue" : ""
+            }`}
+            onClick={() => handleUiClick(filterStates.all)}
+          >
+            <ListIcon height={20} width={20} className={`${filterState === filterStates.all ? "text-events-user-card-bg fill-current" : ""}`} />
+            <p className={`${filterState === filterStates.all ? "text-events-user-card-bg" : ""} text-nowrap`}>All Events</p>
+          </div>
+          <div
+            className={`bg-events-card-bg p-2 h-[46px] rounded-lg flex-row justify-center items-center space-x-1 shrink flex grow lg:hidden cursor-pointer ${
+              filterState === filterStates.saved ? "!bg-events-baby-blue" : ""
+            }`}
+            onClick={() => handleUiClick(filterStates.saved)}
+          >
+            <Bookmark height={20} width={20} className={`${filterState === filterStates.saved ? "text-events-user-card-bg fill-current" : ""}`} />
+            <p className={`${filterState === filterStates.saved ? "text-events-user-card-bg" : ""}`}>Saved</p>
+          </div>
+        </div>
```

Comment:
Two comments:
1. I was having troubles reading this code - at a glance, all of these divs look about the same, with a few minor changes. Would it be possible to improve the readability by an Extract Component refactoring?
2. Is the main difference between these divs for responsive design? Could we use Tailwind Breakpoints to simplify this further

Reply:
Good point! I used composition before as a fast way to build the UI, but extracting components makes sense. I'll also implement breakpoints with flex-basis which should remove some of the redundant code.